regulation-worker/internal/client/client_test.go

```diff
@@ -44,17 +46,31 @@ func TestGet(t *testing.T) {
 			respCode:    429,
 			expectedErr: fmt.Errorf("unexpected response code: 429"),
 		},
+		{
+			name:        "Get request to get model.ErrRequestTimeout",
+			workspaceID: "1001",
+			expectedErr: model.ErrRequestTimeout,
+			serverDelay: 1,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-				w.WriteHeader(tt.respCode)
+				if tt.respCode != 0 {
+					w.WriteHeader(tt.respCode)
+				}
+				time.Sleep(time.Duration(tt.serverDelay) * time.Second)
```

Comment:
Alternatively I think you could set the client timeout to one nanosecond and do a 1ms sleep here to get a faster test.

Not major of course but I'm proposing this because I think that 1s for such a test it's still a long time and can be easily fixed.

Reply:
makes sense. Modified.